tests/pytests/test_multibyte_char_terms.py

```diff
@@ -1434,3 +1441,92 @@ def testTagSearch(env):
 
     res = conn.execute_command('FT.SEARCH', 'idx', '@t:{fussball}', 'NOCONTENT')
     env.assertEqual(res, [1, 'doc:2s'])
+
+def testLongTags(env):
+    env.cmd('FT.CREATE', 'idx', 'SCHEMA', 't', 'TAG')
+    conn = getConnectionByEnv(env)
+
+    t = unescaped_long_term
+    t_lower = t.lower()
+    t_escaped = escaped_long_term
+    t_escaped_lower = t_escaped.lower()
+
+    env.expect('HSET', '{doc}:1', 't', t).equal(1)
+    if not env.isCluster():
+        res = env.cmd(debug_cmd(), 'DUMP_TAGIDX', 'idx', 't')
+        # The conversion to lowercase occupies more space than the original
+        # term, so the term is stored in its original form
+        env.assertEqual(res, [[t, [1]]])
+
+    env.expect('HSET', '{doc}:2', 't', t_lower).equal(1)
+    if not env.isCluster():
+        res = env.cmd(debug_cmd(), 'DUMP_TAGIDX', 'idx', 't')
+        # The lowercase term is stored in its original form, so the index will
+        # have two terms: the original term and the lowercase term
+        env.assertEqual(res, [[t, [1]], [t_lower, [2]]])
+
+    for dialect in range(1, 5):
+        res = conn.execute_command(
+            'FT.SEARCH', 'idx', f'@t:{{{t_escaped}}}', 'NOCONTENT', 'DIALECT', dialect)
+        env.assertEqual(res, [1, '{doc}:1'])
+
+        res = conn.execute_command(
+            'FT.SEARCH', 'idx', f'@t:{{{t_escaped_lower}}}', 'NOCONTENT', 'DIALECT', dialect)
+        env.assertEqual(res, [1, '{doc}:2'])
```

Comment:
Once reallocation is enabled, should each query return both docs?

Reply:
Yes.